install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -2638,18 +2638,80 @@ sctp:
 
 auth:
   mTLS:
-    # -- Enable mtls-spiffe authentication method in CiliumNetworkPolicy
-    enabled: false
-    # -- SPIRE server endpoint
-    # This endpoint will be automatically injected later once embedded SPIRE installation is done.
-    spireServerAddress: spire-server.spire.svc.cluster.local:8081
-    # -- SPIRE server connection timeout
-    spireServerConnectionTimeout: 10s
-    # -- SPIRE agent socket path where the SPIRE agent is listening
-    spireAgentSocketPath: /run/spire/sockets/agent/agent.sock
-    # -- SPIRE socket path where the SPIRE delegated api agent is listening
-    spireAdminSocketPath: /run/spire/sockets/admin.sock
-    # -- SPIFFE trust domain to use for fetching certificates
-    spiffeTrustDomain: spiffe.cilium.io
-    # -- port on the agent which is used to mTLS handshakes on
+    # -- Port on the agent where mTLS handshakes between agents will be performed
     port: 4250
+    # Settings for SPIRE
+    spire:
+      # -- Enable SPIRE integration
+      enabled: false
+      # -- Settings to control the SPIRE installation and configuration
+      install:
+        # -- Enable SPIRE installation.
+        # This will only take effect only if auth.mTLS.spire.enabled is true
+        enabled: false
+        # -- SPIRE namespace to install into
+        namespace: cilium-spire
+        # -- SPIRE agent configuration
+        agent:
+          # -- SPIRE agent image
+          image: ghcr.io/spiffe/spire-agent:1.5.1@sha256:40228af4d9a094f0fef2d7a303a3b6a689c4b4eba2fa9f7da5125b81d2d68ec8
```

Comment:
Are we going to mirror these containers to a repository in the cilium organization?

Reply:
most of users are kind of doing that i.e. copy public images to private registry (e.g. ecr, google, etc), so I think we don't need to do it ourselves.